services/QuillLMS/spec/controllers/teachers_controller_spec.rb

```diff
@@ -19,8 +19,7 @@
       end
 
       it 'render admin dashboard' do
-        user = create(:user)
-        user.schools_admins.create
+        user = create(:user, role: User::ADMIN)
```

Comment:
To keep with the pattern in users.rb, I would create a new factory

```
factory :admin do
  role { User::ADMIN }
end
```

The interesting question is do you embed this inside of the `factory :teacher` block definition.

Then you have `user = create(:admin)` which cuts down on boilerplate and standardizes the factory construction for easier refactoring later.

Reply:
There are a few more of this admin construction in the specs of this in the PR.